datafusion/spark/src/function/math/width_bucket.rs

```diff
@@ -96,7 +96,14 @@ impl ScalarUDFImpl for SparkWidthBucket {
         let is_num = |t: &DataType| {
             matches!(
                 t,
-                Int8 | Int16 | Int32 | Int64 | Float32 | Float64 | Decimal128(_, _)
+                Int8 | Int16
+                    | Int32
+                    | Int64
+                    | Float32
+                    | Float64
+                    | Decimal32(_, _)
+                    | Decimal64(_, _)
+                    | Decimal128(_, _)
```

Comment:
Can use [`is_signed_numeric()`](https://github.com/apache/datafusion/blob/2d947b31d0355710dc179d1d72ca5366b7721b2a/datafusion/expr/src/type_coercion/mod.rs#L43-L59) here potentially (though that brings in `Float16`)